scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -1520,19 +1520,23 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
         }
       case KwTry() =>
         next()
+        val startToken = token
+        def getRestIfNeeded(x: Term) =
+          if (token.is[KwCatch] || token.is[KwFinally]) x
+          else simpleExprRest(atPos(startToken, auto)(x), true)
         val body: Term = token match {
```

Comment:
Judging by the diff in scalac, we need to change how the body is parsed https://github.com/scala/scala/pull/8071/files#diff-e059ce64777235990044f1624042e5d0R1518

<img width="762" alt="Screenshot 2020-04-27 at 08 27 21" src="https://user-images.githubusercontent.com/1408093/80350955-04aeef00-8861-11ea-9b14-1e54871b9d9f.png">

Reply:
done. added link to dialect.